spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/feign/ribbon/FeignLoadBalancer.java

```diff
@@ -74,26 +95,32 @@ public RibbonResponse execute(RibbonRequest request, IClientConfig configOverrid
 		else {
 			options = new Request.Options(this.connectTimeout, this.readTimeout);
 		}
-		Response response = request.client().execute(request.toRequest(), options);
-		return new RibbonResponse(request.getUri(), response);
+		LoadBalancedRetryPolicy retryPolicy = loadBalancedRetryPolicyFactory.create(this.getClientName(), this);
+		retryTemplate.setRetryPolicy(retryPolicy == null ? new NeverRetryPolicy()
```

Comment:
Just trying to make sure I understand the concern....

If multiple requests are executed simultaneously you are concerned about switching out the retry policy mid request?